src/promptflow/promptflow/_core/connection_manager.py

```diff
@@ -44,6 +45,12 @@ def _build_connections(cls, _dict: Dict[str, dict]):
         for key, connection_dict in _dict.items():
             if isinstance(connection_dict, _Connection):
                 # support directly pass connection object to executor
+                scrubbed_secrets = connection_dict._get_scrubbed_secrets()
```

Comment:
what if user has a loca create connection without setting secret?

Reply:
There won't be error since we only check if there are scrubbed secrets in the connection object. Added a test for this: "test_non_secret_connection"